superset/views/base.py

```diff
@@ -424,8 +426,9 @@ def cached_common_bootstrap_data(user: User, locale: str) -> dict[str, Any]:
 
 
 def common_bootstrap_payload(user: User) -> dict[str, Any]:
+    user_id = user.id if hasattr(user, "id") else None
```

Comment:
It seems like the FAB [`User`](https://github.com/dpgaspar/Flask-AppBuilder/blob/master/flask_appbuilder/security/sqla/models.py#L94-L167) class will always have the `id` attribute defined and thus simply passing in `user.id` on line 431 _should_ be safe.

Reply:
Grokking through the [code](https://github.com/search?q=repo%3Aapache%2Fsuperset%20common_bootstrap_payload&type=code) it seems like this is always `g.user`, thus rather than passing around the global, it seems clearer to remove the `user` variable from said method and thus said logic simply becomes, 

```python
 from superset.utils.core import get_user_id 

def common_bootstrap_payload() -> dict[str, Any]:
    return {
        **cached_common_bootstrap_data(get_user_id(), get_locale()),
        "flash_messages": get_flashed_messages(with_categories=True),
    }
```

Reply:
Oh great point it is always `g.user`, I'll make that change.

> It seems like the FAB [`User`](https://github.com/dpgaspar/Flask-AppBuilder/blob/master/flask_appbuilder/security/sqla/models.py#L94-L167) class will always have the `id` attribute defined and thus simply passing in `user.id` on line 431 _should_ be safe.

The type hint was a bit of a lie since `g.user` could also be `AnonymousUserMixin`.  I'd update it but it's getting removed instead

Reply:
@jfrag1 I think my comment is somewhat of a moot point given that it seems like we're going down the `get_user_id()` route.